src/chainstate/stacks/boot/mod.rs

```diff
@@ -0,0 +1,3428 @@
+/*
+ copyright: (c) 2013-2019 by Blockstack PBC, a public benefit corporation.
+
+ This file is part of Blockstack.
+
+ Blockstack is free software. You may redistribute or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License or
+ (at your option) any later version.
+
+ Blockstack is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY, including without the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Blockstack. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use chainstate::stacks::StacksAddress;
+use chainstate::stacks::db::StacksChainState;
+use chainstate::stacks::Error;
+use chainstate::stacks::StacksBlockHeader;
+
+use burnchains::bitcoin::address::BitcoinAddress;
+use burnchains::Address;
+use address::AddressHashMode;
+
+use chainstate::burn::db::sortdb::SortitionDB;
+
+use vm::types::{
+    StandardPrincipalData,
+    QualifiedContractIdentifier,
+    Value,
+    TupleData,
+    PrincipalData,
+    SequenceData
+};
+
+use chainstate::stacks::StacksBlockId;
+
+use burnchains::Burnchain;
+
+use vm::representations::ContractName;
+
+use util::hash::Hash160;
+
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::boxed::Box;
+
+pub const STACKS_BOOT_CODE_CONTRACT_ADDRESS : &'static str = "ST000000000000000000002AMW42H";
+
+pub const STACKS_BOOT_CODE : &'static [(&'static str, &'static str)] = &[
+    ("pox-api", std::include_str!("pox.clar")),     // Clarity prevents us from using anything shorter than 5 characters, so 'pox' isn't an option :(
```

Comment:
`bns` as well :(. should we revisit that contraint? cc @kantai

Reply:
Happy to eliminate that constraint (making min-contract-name-length 1).

Reply:
👍

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1850/commits/4e9abe5ac400e0597b12657325caf62ee4e170e5